manager/logbroker/broker.go

```diff
@@ -28,24 +34,51 @@ type LogBroker struct {
 
 	registeredSubscriptions map[string]*api.SubscriptionMessage
 
-	stopped chan struct{}
+	pctx      context.Context
+	cancelAll context.CancelFunc
 }
 
 // New initializes and returns a new LogBroker
 func New() *LogBroker {
-	return &LogBroker{
-		logQueue:                watch.NewQueue(),
-		subscriptionQueue:       watch.NewQueue(),
-		registeredSubscriptions: make(map[string]*api.SubscriptionMessage),
-		stopped:                 make(chan struct{}),
+	return &LogBroker{}
+}
+
+// Run the log broker
+func (lb *LogBroker) Run(ctx context.Context) error {
+	lb.mu.Lock()
+
+	if lb.cancelAll != nil {
+		lb.mu.Unlock()
+		return errAlreadyRunning
+	}
+
+	lb.pctx, lb.cancelAll = context.WithCancel(ctx)
+	lb.logQueue = watch.NewQueue()
+	lb.subscriptionQueue = watch.NewQueue()
+	lb.registeredSubscriptions = make(map[string]*api.SubscriptionMessage)
+	lb.mu.Unlock()
+
+	select {
+	case <-lb.pctx.Done():
+		return lb.pctx.Err()
 	}
 }
 
 // Stop stops the log broker
-func (lb *LogBroker) Stop() {
-	close(lb.stopped)
+func (lb *LogBroker) Stop() error {
+	lb.mu.Lock()
+	defer lb.mu.Unlock()
+
+	if lb.cancelAll == nil {
+		return errNotRunning
+	}
+	lb.cancelAll()
+	lb.cancelAll = nil
+
 	lb.logQueue.Close()
 	lb.subscriptionQueue.Close()
+
+	return nil
```

Comment:
I think it's good practice for `Stop` to block until all current RPCs are finished. I don't see anything especially dangerous in the RPCs like interactions with Raft, but we may add something in the future. Also, right now it's theoretically possible for an RPC started before a leader reelection to interact with the new queue created by `Run`, which feels wrong.